classes/amity.py

```diff
@@ -0,0 +1,317 @@
+import random
+
+from database.database_connections import DatabaseConnections
+from classes.person import Fellow
+from classes.person import Staff
+from classes.room import Office
+from classes.room import LivingSpace
+
+
+class Amity(object):
+
+    staff = {}
+    fellows = {}
+    offices = {}
+    livingspaces = {}
+
+    def create_room(name, kind):
+        """creates a room depending on the type specified"""
+
+        #check if room is already in the system
+        if(name.lower() in (key.lower() for room in [Amity.offices.keys(), Amity.livingspaces.keys()] for key in room)):
+            return "Room already exists"
+        #create room object and add it to respective dictionary
+        elif kind == "office":
+            Amity.offices[name] = Office(name)
+        elif kind == "livingspace":
+            Amity.livingspaces[name] = LivingSpace(name)
+        return "Room {} has been created".format(name)
+
+    def add_person(fname, lname, title, wants_accommodation=None):
+        """creates a new person basing on their specified type"""
+
+        initial_id = input("Enter {} {}'s staff id: ".format(fname, lname))
+        #calls the validate staff id to ensure it is unique
+        staff_id = Amity.validate_staff_id(initial_id)
+        #check if the person is staff or fellow
+        if title == "FELLOW":
+            person = Fellow(fname, lname, staff_id)
+            Amity.assign_office(person)
+            #if wants accommodation assign livingspace
+            if wants_accommodation == "Y":
+                Amity.assign_livingspace(person)
+            Amity.fellows[person.staff_id] = person
+        if title == "STAFF":
+            person = Staff(fname, lname, staff_id)
+            Amity.assign_office(person)
+            Amity.staff[person.staff_id] = person
+        #determine what to return for office name
+        if person.allocated_office == "":
+            return "{} {} added but not assigned room".format(person.first_name, 
+                                                                person.last_name)
+        else:
+            return "{} {} added and assigned to {}".format(person.first_name, 
+                                                            person.last_name, 
+                                                            person.allocated_office)
+
+    def validate_staff_id(staff_id):
+        """determines that the staff id is not in the system"""
+
+        if staff_id in Amity.fellows.keys() or staff_id in Amity.staff.keys():
```

Comment:
Thank you, did not know it iterates over the keys and not the values, much appreciated